src/http/one/RequestParser.cc

```diff
@@ -14,11 +14,17 @@
 #include "SquidConfig.h"
 
 Http1::Parser::size_type
-Http::One::RequestParser::firstLineSize() const
+Http::One::RequestParser::FirstLineSize(const size_t methodLength, const size_t uriLength)
```

Comment:
This is no longer about just parsing. Can we "move" this _and_ the corresponding request method without heroic efforts?

```suggestion
Http::One::FirstLineSize(const size_t methodLength, const size_t requestTargetLength)
```

If we can convert these static methods into free functions, I would still leave Http::One::RequestParser::firstLineSize() where it is now -- that method is specific to the parser.